### PR TITLE
Fixes for set_uid and set_gid for multi threaded

### LINKED ISSUE #6467: Endpoint pod crashes when running warp
### Environment info
- NooBaa Version: 
[root@curlicue-inf ~]# noobaa version
INFO[0000] CLI version: 5.8.0
INFO[0000] noobaa-image: noobaa/noobaa-core:5.8.0-20210418
INFO[0000] operator-image: noobaa/noobaa-operator:5.8.0
[root@curlicue-inf ~]# oc version
Client Version: 4.6.18
Server Version: 4.6.18
Kubernetes Version: v1.19.0+f173eb4
[root@curlicue-inf ~]#

### Actual behavior
1. Run warp using
warp mixed --access-key=<access-key>--secret-key=<secretkey> --obj.size=1M --host=s3-noobaa.apps.curlicue.os.fyre.ibm.com  --duration=60s --objects=10 --bucket=fs1-s3user3-bucket  --debug --tls --insecure
2. During  warp run get 503 service error as endpoint pod crashes
 warp: <ERROR> upload error: 503 Service Unavailable
warp: <ERROR> upload error: 503 Service Unavailable
warp: <ERROR> upload error: 503 Service Unavailable
warp: <ERROR> upload error: 503 Service Unavailable

### Expected behavior
1. The endpoint should not crash and warp should run successfully

### Steps to reproduce
1. Followed all step from https://github.com/noobaa/noobaa-core/wiki/NSFS-deployment-with-export-of-accounts-(UID,-GID-configuration)
2. Run warp on the bucket created

### More information - Screenshots / Logs / Other output
Endpoint pod and core pod logs attached
[endpoint.log](https://github.com/noobaa/noobaa-core/files/6362898/endpoint.log)
[core.log](https://github.com/noobaa/noobaa-core/files/6362903/core.log)

### PR DESCRIPTION
**Pull request by jackyalbo**
Signed-off-by: jackyalbo <jacky.albo@gmail.com>

### Explain the changes
1.  changed the function calls of seteuid and setegid to system calls SYS_setresuid SYS_setresgid respectively in order to better support multi threaded runs
2. change the order of setuid and setgid after the work finish as setgid should come before setuid.
3. added panic calls in case where running setgid/setuid failes
4. added extra prints and info for some of the functions in fs_napi

### Issues: Fixed #xxx / Gap #xxx
1.  partially Fixed #6467 - fully fixed together with a future PR from @guymguym 

### Testing Instructions:
1.

### CODE REVIEW

File: src/native/fs/fs_napi.cpp [hunk 1]
```diff
@@ -149,6 +159,22 @@ struct FSWorker : public Napi::AsyncWorker
             std::string errmsg = strerror(errno);
             SetError(errmsg);
         }
+        bool change_uid = orig_uid != _req_uid;
+        bool change_gid = orig_gid != _req_gid;
+        if (change_uid) {
+            int r = syscall(SYS_setresuid, -1, orig_uid, -1);
```
Comment (guymguym) [inline src/native/fs/fs_napi.cpp:165]: Why is this setuid/setgid needed here? Seems very unrelated to me

File: src/native/fs/fs_napi.cpp [hunk 2]
```diff
@@ -182,6 +208,7 @@ struct Stat : public FSWorker
     }
     virtual void Work()
     {
+        DBG1("FS::Stat::Work: " << DVAL(_path));
```
Comment (guymguym) [inline src/native/fs/fs_napi.cpp:211]: If you want all fsworkers to log1 then add a print on the base class in Execute...

**Review by guymguym — CHANGES_REQUESTED** [review]
Two comments

File: src/native/fs/fs_napi.cpp [hunk 3]
```diff
@@ -149,6 +159,22 @@ struct FSWorker : public Napi::AsyncWorker
             std::string errmsg = strerror(errno);
             SetError(errmsg);
         }
+        bool change_uid = orig_uid != _req_uid;
```
Comment (jeniawhite) [inline src/native/fs/fs_napi.cpp:162]: SetSyscallError can be called multiple times in a single run.
In this case, you will set to orig_uid and orig_gid multiple times.

File: src/native/fs/fs_napi.cpp (hunk 1 above)
Comment (jackyalbo) [reply src/native/fs/fs_napi.cpp:165]: right, now that it panics - this one can be removed

File: src/native/fs/fs_napi.cpp (hunk 3 above)
Comment (jackyalbo) [reply src/native/fs/fs_napi.cpp:162]: will be removed

File: src/native/fs/fs_napi.cpp (hunk 2 above)
Comment (guymguym) [reply src/native/fs/fs_napi.cpp:189]: But the base class prints the action name using _desc... Not sure why you want to leave all those calls when you already have it?

Comment (jackyalbo) [reply src/native/fs/fs_napi.cpp:189]: where do we print the action name? I never seen any print like it. in _desc we have just the path: ```FS::FSWorker::Begin: _path=/nsfs``` and that's for lucky few that has begin.

Comment (guymguym) [reply src/native/fs/fs_napi.cpp:189]: So add the action name to Begin, and add missing calls to Begin from FSWorkers that did not add it.

Comment (jackyalbo) [reply src/native/fs/fs_napi.cpp:189]: #6497

**Pull request merged**